### PR TITLE
[CI] Add jazzy and other minor updates

### PR DESCRIPTION
**Pull request by christophfroehlich**
- Add Jazzy workflows (I decided to create dedicated files, because it easier to maintain if we branch for jazzy)
- Add filters to run the ROS - workflows only if code has changed, and the doc workflow only if docs have changed. (branches+path filter have to match both to run the workflow)
- Check Docs - workflow now uses the correct branch of control.ros.org (not every page is available on every distro any more).
- Run pre-commit auto-update only once per month

### CODE REVIEW

File: .github/workflows/humble-binary-build.yml [hunk 1]
```diff
@@ -7,9 +7,19 @@ on:
   pull_request:
     branches:
       - humble
+    paths:
+      - '**.hpp'
+      - '**.h'
+      - '**.cpp'
+      - '.github/workflows/humble-binary-build.yml'
```
Comment (fmauch) [inline .github/workflows/humble-binary-build.yml:14]: It probably also makes sense to run on changes to 

- package.xml
- CMakeLists.txt
- repo files

That comment would be relevant for all build jobs.

File: .github/workflows/iron-rhel-semi-binary-build.yml [hunk 2]
```diff
@@ -4,6 +4,11 @@ on:
   pull_request:
     branches:
       - iron
+    paths:
+      - '**.hpp'
+      - '**.h'
+      - '**.cpp'
+      - '.github/workflows/jazzy-debian-build.yml'
```
Comment (fmauch) [inline .github/workflows/iron-rhel-semi-binary-build.yml:11]:
```suggestion
      - '.github/workflows/iron-rhel-semi-binary-build.yml'
```

**Review by fmauch — CHANGES_REQUESTED** [review]

**Comment by fmauch** [comment]
Also, I was wondering, why the check_docs workflows didn't have any path restrictions, now I am wondering even more while reviewing https://github.com/ros-controls/ros2_control/pull/1521

File: .github/workflows/humble-binary-build.yml (hunk 1 above)
Comment (christophfroehlich) [reply .github/workflows/humble-binary-build.yml:14]: you are right, I'll check this

**Review by fmauch — APPROVED** [review]
LGTM. Thanks for all the work @christophfroehlich 

The failing CI seems unrelated.

**Pull request merged**